code/scripting/api/objs/object.cpp

```diff
@@ -541,17 +541,9 @@ ADE_FUNC(assignSound, l_Object, "soundentry GameSnd, [vector Offset=nil, enumera
 	auto subsys = tgsh ? tgsh->ss : nullptr;
 	if (!offset)
 		offset = &vmd_zero_vector;
-	if (enum_flags.IsValid())
-	{
+	if (enum_flags.index >= 0)
```

Comment:
I can't think of a way to generate the "0" value for this. Maybe it would be good to also add a `OS_NONE` enum with the value 0 so that can be specified here if not flags are desired.

Reply:
That's a good point.